tf_encrypted/config.py

```diff
@@ -17,290 +18,296 @@
+  cfs_quota = Path("/sys/fs/cgroup/cpu/cpu.cfs_quota_us")
 
-    if cfs_period.exists() and cfs_quota.exists():
-        with cfs_period.open('rb') as p, cfs_quota.open('rb') as q:
-            p_int, q_int = int(p.read()), int(q.read())
+  if cfs_period.exists() and cfs_quota.exists():
+    with cfs_period.open('rb') as p, cfs_quota.open('rb') as q:
+      p_int, q_int = int(p.read()), int(q.read())
 
-            # get the cores allocated by dividing the quota
-            # in microseconds by the period in microseconds
-            if q_int > 0 and p_int > 0:
-                cpu_cores = math.ceil(q_int / p_int)
+      # get the cores allocated by dividing the quota
+      # in microseconds by the period in microseconds
+      if q_int > 0 and p_int > 0:
+        cpu_cores = math.ceil(q_int / p_int)
 
-    return cpu_cores
+  return cpu_cores
 
 
 class Config(ABC):
+  """The main tfe.Config abstraction."""
+
+  @abstractmethod
+  def players(self):
+    """Returns the config's list of :class:`Player` objects."""
+
+  @abstractmethod
+  def get_player(self, name):
+    """Retrieve a specific :class:`Player` object by name."""
+
+  @abstractmethod
+  def get_tf_config(
+      self,
+      log_device_placement=False,
+      disable_optimizations=False
+  ):
+    """
+    get_tf_config(log_device_placement=False) -> tf.ConfigProto, or str
 
-    @abstractmethod
-    def players(self):
-        """
-        Returns the config's list of :class:`Player` objects.
-        """
-        pass
-
-    @abstractmethod
-    def get_player(self, name):
-        """
-        Retrieve a specific :class:`Player` object by name.
-        """
-        pass
-
-    @abstractmethod
-    def get_tf_config(self, log_device_placement=False, disable_optimizations=False):
-        """
-        get_tf_config() -> tf.ConfigProto, or str
-
-        Extract the underlying :class:`tf.ConfigProto`.
-        """
-        pass
-
-    def build_graph_options(self, disable_optimizations):
-        if not disable_optimizations:
-            return tf.GraphOptions()
-        else:
-            return tf.GraphOptions(
-                optimizer_options=tf.OptimizerOptions(
-                    opt_level=tf.OptimizerOptions.L0,
-                    do_common_subexpression_elimination=False,
-                    do_constant_folding=False,
-                    do_function_inlining=False,
-                ),
-                rewrite_options=rewriter_config_pb2.RewriterConfig(
-                    arithmetic_optimization=rewriter_config_pb2.RewriterConfig.OFF,
-                ),
-            )
+    Extract the underlying :class:`tf.ConfigProto`.
+    """
 
+  @classmethod
+  def build_graph_options(cls, disable_optimizations):
+    if not disable_optimizations:
+      return tf.GraphOptions()
 
-class LocalConfig(Config):
-    """
-    Configure TF Encrypted to use threads on the local CPU to simulate the different players.
+    return tf.GraphOptions(
```

Comment:
too much indentation?

Reply:
By convention, 4 space is used when continuing an open parenthesis in order to disambiguate it from a normal indentation.  For example, see [here](https://github.com/tensorflow/tensorflow/blob/a6003151399ba48d855681ec8e736387960ef06e/tensorflow/python/keras/layers/core.py#L94).